### ISSUE TITLE
Select first non-hidden tag on f/e open

### ISSUE THREAD

**Issue by mwinokan** [m2ms/fragalysis-frontend#1481]
Currently, if the first CanonicalSites tag is hidden it will still be selected by default when opening Fragalysis. This can be confusing as the hits in the hit navigator will be shown and hard to hide.

The workaround is to toggle "Select all tags" on and then off again.

@boriskovar-m2ms / @matej-vavrek is it easy for you to change the frontend to open with the first non-hidden tag selected instead? Or do we need to wait for @kaliif to return from his holidays?

**Comment by boriskovar-m2ms** [comment]
@mwinokan I don't think this is hard. But this request made me think a little bit what should be the behavior in case like following:
1. turn some stuff on on LHS
2. create snapshot
3. someone comes in and hides all canonical sites that the hits you made visible in step 1 hidden.
4. Someone access the url that was created in step 2
5. What should happen in this case?

Still knees deep in the snapshots realm so now see snapshots related problems in every proposed feature/fix

**Comment by mwinokan** [comment]
Thanks @boriskovar-m2ms that's a useful insight.

Even though the tag navigator UI might now not show the full selection of tags in the case you describe, the snapshot should still restore the same selection of tags (and therefore hits). 

Ideally, there would be some warning bubbles appear to say something along the lines: "CanonSite X is selected but hidden"

Let me know if that's not clear
